src/wallet/wallet.cpp

```diff
@@ -4061,3 +4061,14 @@ bool CMerkleTx::AcceptToMemoryPool(const CAmount& nAbsurdFee, CValidationState&
     return ::AcceptToMemoryPool(mempool, state, tx, nullptr /* pfMissingInputs */,
                                 nullptr /* plTxnReplaced */, false /* bypass_limits */, nAbsurdFee);
 }
+
+bool CWallet::IsSolvable(const CScript& script) const
+{
+    // This check is to make sure that the script we created can actually be solved for and signed by us
+    // if we were to have the private keys. This is just to make sure that the script is valid and that,
+    // if found in a transaction, we would still accept and relay that transaction. In particular,
+    // it will reject witness outputs that require signing with an uncompressed public key.
+    DummySignatureCreator creator(this);
```

Comment:
Can't this pass in nullptr and move out of CWallet? Unless I'm missing something, it's not actually related.

Reply:
Moved to script/signer.